tests/unistd/truncate.c

```diff
@@ -99,6 +108,15 @@ int main() {
   printf("st_size: %lld\n", s.st_size);
   memset(&s, 0, sizeof s);
   errno = 0;
+  printf("\n");
+
+  printf("open(readonly, O_TRUNC)\n");
+  open("toread", O_RDONLY | O_TRUNC);
```

Comment:
Is this not the same as above? Both open in read-only mode + trunc. Should the former be read-write or write-only?

Reply:
It is the same and that is intentional. The behavior should depend on the permissions of the file, not the mode we are opening it in.

Reply:
I see, sounds good. Maybe add a comment here for that?